src/components/solr-faceted-search.js

```diff
@@ -33,7 +33,7 @@ class SolrFacetedSearch extends React.Component {
     const resultPending = results.pending ? (<ResultPendingComponent bootstrapCss={bootstrapCss}/>) : null;
 
     const pagination = query.pageStrategy === "paginate" ?
-      <PaginateComponent {...this.props} bootstrapCss={bootstrapCss} onChange={onPageChange}/> :
+      <PaginateComponent {...this.props} bootstrapCss={bootstrapCss} onChange={onPageChange} setView={onSetView}/> :
```

Comment:
Pass the function that allows the paginate component to use the redux store. At the redux store, we save the page that we are currently on. This helps us to determine whether we want to update the component in `iSamples_resultList`.